public/javascripts/index.js

```diff
@@ -159,11 +198,50 @@ function deleteResources(url, successCallback, errorCallback) {
     accepts: {
       text: "application/hal+json"
     },
-    success: function(data, textStatus, jQxhr) {
+    success: function() {
       successCallback();
     },
-    error: function(jqXhr, textStatus, errorThrown) {
+    error: function(jqXhr) {
       errorCallback(jqXhr.responseJSON);
     }
   });
 }
+
+function buildMaterialMenuItems(clickedElementData) {
+  const baseOptions = [
+    {
+      type: "normal",
+      text: "Delete pacts ...",
+      click: handleDeletePactsSelected
+    },
+    {
+      type: "normal",
+      text: "Delete integration...",
+      click: handleDeleteIntegrationsSelected
+    }
+  ];
+
+  const providerName = clickedElementData.providerName;
+  const consumerName = clickedElementData.consumerName;
+  const taggedPactUrls = clickedElementData.taggedPactUrls;
+  const taggedPactsOptions = taggedPactUrls.map(taggedPactUrl => {
+    const pactTagName = getTagNameFromTaggedPactUrl(taggedPactUrl);
+    return {
+      type: "normal",
+      text: `Delete pact tagged ${pactTagName}...`,
+      click: handleDeleteTagSelected({
+        providerName,
+        consumerName,
+        pactTagName,
+        deletionUrl: taggedPactUrl
+      })
+    };
+  });
+
+  return [...baseOptions, ...taggedPactsOptions];
+}
+
+function getTagNameFromTaggedPactUrl(taggedPactUrl) {
```

Comment:
Is there a way to pass in structured data so we don't need to do this? It's better than manually constructing the URL in the JS, but it still does mean the UI has to know how the URL is constructed, and that feels a bit icky. Can we put JSON in the data?

Reply:
Fixed in db229f34 by returning a JSON in below format

```json
{
  "tag":"tagName",
  "deletionUrl": "deletionUrl"
}
```

Reply:
Cool!